src/base/rss/private/rss_parser.cpp

```diff
@@ -391,20 +391,22 @@ namespace
         int nmin   = 8;
         int nsec   = 9;
         // Also accept obsolete form "Weekday, DD-Mon-YY HH:MM:SS ±hhmm"
-        QRegExp rx("^(?:([A-Z][a-z]+),\\s*)?(\\d{1,2})(\\s+|-)([^-\\s]+)(\\s+|-)(\\d{2,4})\\s+(\\d\\d):(\\d\\d)(?::(\\d\\d))?\\s+(\\S+)$");
+        static const QRegularExpression rx("^(?:([A-Z][a-z]+),\\s*)?(\\d{1,2})(\\s+|-)([^-\\s]+)(\\s+|-)(\\d{2,4})\\s+(\\d\\d):(\\d\\d)(?::(\\d\\d))?\\s+(\\S+)$");
         QStringList parts;
-        if (!str.indexOf(rx)) {
+        QRegularExpressionMatch rxMatch;
+        if (!str.indexOf(rx, 0, &rxMatch)) {
             // Check that if date has '-' separators, both separators are '-'.
-            parts = rx.capturedTexts();
+            parts = rxMatch.capturedTexts();
             bool h1 = (parts[3] == QLatin1String("-"));
             bool h2 = (parts[5] == QLatin1String("-"));
             if (h1 != h2)
                 return QDateTime::currentDateTime();
         }
         else {
             // Check for the obsolete form "Wdy Mon DD HH:MM:SS YYYY"
-            rx = QRegExp("^([A-Z][a-z]+)\\s+(\\S+)\\s+(\\d\\d)\\s+(\\d\\d):(\\d\\d):(\\d\\d)\\s+(\\d\\d\\d\\d)$");
-            if (str.indexOf(rx))
+            static const QRegularExpression rx2("^([A-Z][a-z]+)\\s+(\\S+)\\s+(\\d\\d)\\s+(\\d\\d):(\\d\\d):(\\d\\d)\\s+(\\d\\d\\d\\d)$");
+            QRegularExpressionMatch rx2Match;
+            if (str.indexOf(rx2, 0, &rx2Match))
```

Comment:
Also please fix this (and below) invalid usage int in boolean context.
Why not use QRegularExpression::match() instead of QString::indexOf()?

Reply:
>Also please fix this (and below) invalid usage int in boolean context.

I'll change it to `if (str.indexOf(rx2, 0, &rx2Match) > 0)` (and below of course).

>Why not use QRegularExpression::match() instead of QString::indexOf()?

I'm not sure what it is doing here... seems it is trying to find an index that is >0... are you sure changing it with QRegularExpressionMatch::hasMatch() is ok?

Reply:
>are you sure changing it with QRegularExpressionMatch::hasMatch() is ok?

Yes. Since the pattern has both `^` and `$` anchors there is only one match is possible.

Reply:
>seems it is trying to find an index that is `> 0`

`if (str.indexOf(rx))` means `if (str.indexOf(rx) != 0)`
`if (!str.indexOf(rx))` means `if (str.indexOf(rx) == 0)`

Reply:
Or maybe I should avoid touching this function? the comment says: Ported to Qt from KDElibs4.
And really you need such custom parser for RSS dates? QDateTime can't handle it?

Reply:
>And really you need such custom parser for RSS dates? QDateTime can't handle it?

I didn't go into that in detail.
>Or maybe I should avoid touching this function? 

Ok, if you unsure. I'll try to see it later.

Reply:
>Ok, if you unsure. I'll try to look it later.

reverted.